commit 528cd0b45e58df1041e58a8ce7453808b44836b7
Author: Andres Salomon <dilinger@queued.net>
Date:   Mon Sep 26 23:16:29 2022 -0400

    Fix non-neon ARM builds of libaom
    
    Linking fails when building chromium for non-NEON ARM w/ the following:
    
    ld.lld: error: undefined symbol: aom_arm_cpu_caps
    >>> referenced by av1_rtcd.h:1055 (../../third_party/libaom/source/config/linux/arm/config/av1_rtcd.h:1055)
    >>>               libaom/av1_rtcd.o:(.text+0x0) in archive obj/third_party/libaom/libaom.a
    >>> referenced by aom_dsp_rtcd.h:3476 (../../third_party/libaom/source/config/linux/arm/config/aom_dsp_rtcd.h:3476)
    >>>               libaom/aom_dsp_rtcd.o:(.text+0x0) in archive obj/third_party/libaom/libaom.a
    >>> referenced by aom_scale_rtcd.h:162 (../../third_party/libaom/source/config/linux/arm/config/aom_scale_rtcd.h:162)
    >>>               libaom/aom_scale_rtcd.o:(.text+0x0) in archive obj/third_party/libaom/libaom.a
    
    The function aom_arm_cpu_caps() in third_party/libaom is used inside
    of several libaom header files, and is used by all arm boards to
    actually determine the running CPU capabilities. As such,
    arm_cpudetect.c needs to be included in sources for all arm boards,
    not just ones that support NEON.

--- a/third_party/libaom/BUILD.gn
+++ b/third_party/libaom/BUILD.gn
@@ -197,9 +197,7 @@ if (current_cpu == "arm64" || cpu_arch_f
     }
     configs += [ ":libaom_config" ]
 
-    # https://bugs.chromium.org/p/aomedia/issues/detail?id=2294
-    sources = [ "//third_party/libaom/source/libaom/aom_ports/arm_cpudetect.c" ]
-    sources += aom_av1_common_intrin_neon
+    sources = aom_av1_common_intrin_neon
     sources += aom_dsp_common_intrin_neon
     sources += aom_av1_encoder_intrin_neon
     sources += aom_dsp_encoder_intrin_neon
@@ -256,6 +254,10 @@ static_library("libaom") {
       cpu_arch_full == "arm-neon-cpu-detect") {
     deps += [ ":libaom_intrinsics_neon" ]
   }
+  if (current_cpu == "arm64" || current_cpu == "arm") {
+    # This is needed by all arm boards due to aom_arm_cpu_caps()
+    sources += [ "//third_party/libaom/source/libaom/aom_ports/arm_cpudetect.c" ]
+  }
   if (is_android) {
     deps += [ "//third_party/android_ndk:cpu_features" ]
   }
